vm/src/builtins/set.rs

```diff
@@ -346,11 +347,41 @@ impl PySetInner {
         vm: &VirtualMachine,
     ) -> PyResult<()> {
         for iterable in others {
+            self.update_internal(iterable, vm)?;
+        }
+        Ok(())
+    }
+
+    fn update_internal(&self, iterable: ArgIterable, vm: &VirtualMachine) -> PyResult<()> {
+        let obj = iterable.to_owned().into_object();
+
+        // check AnySet
+        if let Ok(_any_set) = AnySet::try_from_object(vm, obj.to_owned()) {
+            self.set_merge(obj, vm)
+        } else {
+            // need to check Dict
+
+            // add items that is not AnySet
             for item in iterable.iter(vm)? {
                 self.add(item?, vm)?;
             }
+            Ok(())
+        }
+    }
+
+    fn set_merge(&self, any_set: PyObjectRef, vm: &VirtualMachine) -> PyResult<()> {
```

Comment:
```suggestion
    fn set_merge(&self, any_set: AnySet, vm: &VirtualMachine) -> PyResult<()> {
```
Once we checked the type is AnySet, keeping the type is better to encode the information into the code.
We might add `fn as_inner()` to `AnySet` for better readability.